firebase-firestore/src/main/java/com/google/firebase/firestore/AggregateField.java

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RestrictTo;
+import java.util.Objects;
+
+// TODO(sumavg): Remove the `hide` and scope annotations.
+/** @hide */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+public abstract class AggregateField {
+  @Nullable private final FieldPath fieldPath;
+
+  @NonNull private final String operator;
+
+  @NonNull private final String alias;
+
+  private AggregateField(@Nullable FieldPath fieldPath, @NonNull String operator) {
+    this.fieldPath = fieldPath;
+    this.operator = operator;
+    // Use $operator_$field format if it's an aggregation of a specific field. For example: sum_foo.
+    // Use $operator format if there's no field. For example: count.
+    this.alias = operator + (fieldPath == null ? "" : "_" + fieldPath);
+  }
+
+  /**
+   * Returns the field on which the aggregation takes place. Returns an empty string if there's no
+   * field (e.g. for count).
+   */
+  @NonNull
+  public String getFieldPath() {
+    return fieldPath == null ? "" : fieldPath.toString();
+  }
+
+  /** Returns the alias used internally for this aggregate field. */
+  @RestrictTo(RestrictTo.Scope.LIBRARY)
+  @NonNull
+  public String getAlias() {
```

Comment:
We need to escape the alias to support the case where it includes special characters. Consider implementing the [AggregateAlias](https://github.com/firebase/firebase-js-sdk/blob/83740c08c0597c3f47ccbeb9afec6ebfa7ad753a/packages/firestore/src/model/aggregate_alias.ts) class to help with this.

Reply:
We're waiting for clarification on the supported alias formats. Otherwise, the new approach looks good.

Reply:
Not sure if you saw the conversation yesterday, but we decided to revert this change to the alias behavior. We can go back to "<op>_<escapedFieldPath>

Reply:
fwiw, in Android it will be `op_encodedPath`, since the string format of fieldPath is the same as encoded path

Reply:
TY